src/averageAges.js

```diff
@@ -14,12 +14,18 @@
  *
  * @return {number}
  */
+// learn how to use array methods like .filter .map .some .every .find .reduce
+// avoid using loop and forEach
+// replace `if ()` statement with &&, || or ?:
+// without nesting
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const arrayOfMen = people.filter(man => man.sex === 'm');
+  const arrayOfMenByCenturies = arrayOfMen.filter(man => century
+    ? Math.ceil(man.died / 100) === century : man);
```

Comment:
think you can try optimizing your code here like in the last function

 `  const peopleWithMother = people.filter(person => people.some(mother => (
    mother.name === person.mother && (onlyWithSon ? person.sex === 'm' : person)
  )));`